cpu/sam3x8e/periph/pwm.c

```diff
@@ -0,0 +1,319 @@
+    PWM_0_DEV_CH6->PWM_CDTY = 0;
+    PWM_0_DEV->PWM_ENA = PWM_0_ENA_CH6;
+#endif
+#if PWM_0_CHANNELS > 7
+    PWM_0_DEV_CH7->PWM_CMR = PWM_CMR_CPRE_CLKA;
+    PWM_0_DEV_CH7->PWM_CPRD = resolution - 1;
+    PWM_0_DEV_CH7->PWM_CDTY = 0;
+    PWM_0_DEV->PWM_ENA = PWM_0_ENA_CH7;
+#endif
+
+    /* +++++++++++ Configure and init channels ready++++++++++ */
+
+
+    /*
+     * Disable GPIO and set peripheral A/B ("0/1") for pwm channel pins.
+     *
+     * TODO: Find out why simple assignment '=' isn't working although
+     * registers are Write only
+     */
+#if PWM_0_CHANNELS > 0
+    PWM_0_PORT_CH0->PIO_PDR  |= PWM_0_PIN_CH0;
+    PWM_0_PORT_CH0->PIO_ABSR |= PWM_0_PIN_CH0;
+#endif
+#if PWM_0_CHANNELS > 1
+    PWM_0_PORT_CH1->PIO_PDR  |= PWM_0_PIN_CH1;
+    PWM_0_PORT_CH1->PIO_ABSR |= PWM_0_PIN_CH1;
+#endif
+#if PWM_0_CHANNELS > 2
+    PWM_0_PORT_CH2->PIO_PDR  |= PWM_0_PIN_CH2;
+    PWM_0_PORT_CH2->PIO_ABSR |= PWM_0_PIN_CH2;
+#endif
+#if PWM_0_CHANNELS > 3
+    PWM_0_PORT_CH3->PIO_PDR  |= PWM_0_PIN_CH3;
+    PWM_0_PORT_CH3->PIO_ABSR |= PWM_0_PIN_CH3;
+#endif
+#if PWM_0_CHANNELS > 4
+    PWM_0_PORT_CH4->PIO_PDR  |= PWM_0_PIN_CH4;
+    PWM_0_PORT_CH4->PIO_ABSR |= PWM_0_PIN_CH4;
+#endif
+#if PWM_0_CHANNELS > 5
+    PWM_0_PORT_CH5->PIO_PDR  |= PWM_0_PIN_CH5;
+    PWM_0_PORT_CH5->PIO_ABSR |= PWM_0_PIN_CH5;
+#endif
+#if PWM_0_CHANNELS > 6
+    PWM_0_PORT_CH6->PIO_PDR  |= PWM_0_PIN_CH6;
+    PWM_0_PORT_CH6->PIO_ABSR |= PWM_0_PIN_CH6;
+#endif
+#if PWM_0_CHANNELS > 7
+    PWM_0_PORT_CH7->PIO_PDR  |= PWM_0_PIN_CH7;
+    PWM_0_PORT_CH7->PIO_ABSR |= PWM_0_PIN_CH7;
+#endif
+
+    return retval;
+}
+
+/*!
+ * Update duty-cycle in channel with \value.
+ * If value is larger than resolution set by pwm_init() it is cropped.
+ */
+int pwm_set(pwm_t dev, int channel, unsigned int value)
+{
+
+    int retval = ERR_SET_CHAN;      /* Worst case */
+    uint32_t period = 0;            /* Store pwm period */
+    PwmCh_num *chan = (void *)0;    /* Addressed channel. */
+
+
+    switch (channel) {
+#if PWM_0_CHANNELS > 0
+
+        case 0:
+            chan = PWM_0_DEV_CH0;
+            break;
+#endif
+#if PWM_0_CHANNELS > 1
+
+        case 1:
+            chan = PWM_0_DEV_CH1;
+            break;
+#endif
+#if PWM_0_CHANNELS > 2
+
+        case 2:
+            chan = PWM_0_DEV_CH2;
+            break;
+#endif
+#if PWM_0_CHANNELS > 3
+
+        case 3:
+            chan = PWM_0_DEV_CH3;
+            break;
+#endif
+#if PWM_0_CHANNELS > 4
+
+        case 4:
+            chan = PWM_0_DEV_CH4;
+            break;
+#endif
+#if PWM_0_CHANNELS > 5
+
+        case 5:
+            chan = PWM_0_DEV_CH5;
+            break;
+#endif
+#if PWM_0_CHANNELS > 6
+
+        case 6:
+            chan = PWM_0_DEV_CH6;
+            break;
+#endif
+#if PWM_0_CHANNELS > 7
+
+        case 7:
+            chan = PWM_0_DEV_CH7;
+            break;
+#endif
+
+        default:
+            ;
+    }
+
+    if (chan) {
+        period = chan->PWM_CPRD & PWM_MAX_VALUE;
```

Comment:
Which value is it that you read here? I'm just wondering about the `if` statement some lines below